server/src/main/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapper.java

```diff
@@ -164,13 +163,43 @@ public Builder(String name, IndexVersion indexVersionCreated) {
             });
         }
 
+        public Builder(String name, IndexVersion indexVersionCreated, int dims) {
+            super(name);
+
+            final boolean indexedByDefault = indexVersionCreated.onOrAfter(INDEXED_BY_DEFAULT_INDEX_VERSION);
+            if (indexedByDefault == false) {
+                throw new UnsupportedOperationException("Dynamic dim size is not supported for index version " + indexVersionCreated);
+            }
```

Comment:
If I index a document with let's say an array of 256 floats on a < 8.11 instance, won't this throw an exception? Looks like `postProcessDynamicMappers()` will run and attempt to dynamically map the dense vector.